### PR TITLE
Add `remove` to `external`

### PR DESCRIPTION
**Pull request by haru-44**
The `gmpy2.remove` function is almost the same as `multiplicity`. The `multiplicity` implementation has been moved to `remove`.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/external/ntheory.py [hunk 1]
```diff
@@ -50,6 +50,34 @@ def bit_scan0(x, n=0):
...
+    if f == 2:
+        b = bit_scan1(x)
+        return x >> b, b
+    m = 0
+    y, rem = divmod(x, f)
+    while not rem:
+        x = y
+        m += 1
+        if m > 5:
+            pow_list = [f**2]
```
Comment (smichr) [inline sympy/external/ntheory.py:67]: This is an interesting variation of what we had before but it now involves a list. Is there a reason to prefer this over to keeping track of the scalar `e`? One thing I see is that you only work with squares while the other code worked with even exponents.

Comment (haru-44) [reply sympy/external/ntheory.py:67]: The current version calls `multiplicity` recursively. To stop the recursion, I use the list.

Also, the current version uses only powers of 2. The `e` will never be 6, 10, etc.

Comment (smichr) [reply sympy/external/ntheory.py:67]: Thanks.

**Comment by haru-44** [comment]
thanks

**Pull request merged**